File: content/snippets/python/interview_lcci/05/03_flip_bit_to_win.py
def flip_bit(num: int) -> int:
    # Treat as 32-bit unsigned integer
    if num & 0xFFFFFFFF == 0xFFFFFFFF:
        return 32
    num = num & 0xFFFFFFFF
    cur = prev = 0
    best = 1
    while num > 0:
        if num & 1:
            cur += 1
        else:
            prev = cur
            cur = 0
        best = max(best, prev + 1 + cur)
        num >>= 1
    return best


def main() -> None:
    print(flip_bit(1775))
    print(flip_bit(7))


if __name__ == "__main__":
    main()

File: content/snippets/go/interview_lcci/05/03_flip_bit_to_win.go
package main

import "fmt"

func flipBit(num uint32) int {
	if num == 0xFFFFFFFF {
		return 32
	}
	cur, prev, best := 0, 0, 1
	for num > 0 {
		if num&1 == 1 {
			cur++
		} else {
			prev = cur
			cur = 0
		}
		if v := prev + 1 + cur; v > best {
			best = v
		}
		num >>= 1
	}
	return best
}

func main() {
	fmt.Println(flipBit(1775))
	fmt.Println(flipBit(7))
}
